RecoHI/HiJetAlgos/plugins/HiPuRhoProducer.h

```diff
@@ -0,0 +1,125 @@
+#ifndef RecoHI_HiJetAlgos_HiPuRhoProducer_h
+#define RecoHI_HiJetAlgos_HiPuRhoProducer_h
+
+#include "DataFormats/Candidate/interface/CandidateFwd.h"
+#include "DataFormats/Common/interface/Ptr.h"
+#include "DataFormats/HcalDetId/interface/HcalDetId.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+
+#include "fastjet/ClusterSequence.hh"
+#include "fastjet/JetDefinition.hh"
+#include "fastjet/PseudoJet.hh"
+
+#include <map>
+#include <vector>
+
+//
+// class declaration
+//
+
+class HiPuRhoProducer : public edm::EDProducer {
```

Comment:
make this thread-aware
https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes
at a glance, edm::stream should be the easiest